src/please.go

```diff
@@ -958,11 +958,15 @@ func Please(targets []core.BuildLabel, config *core.Configuration, shouldBuild,
 }
 
 func runPlease(state *core.BuildState, targets []core.BuildLabel) {
-	// Acquire the lock before we start building
-	if (state.NeedBuild || state.NeedTests) && !opts.FeatureFlags.NoLock {
-		core.AcquireRepoLock()
-		defer core.ReleaseRepoLock()
-	}
+	// Every plz instance gets a shared repo lock which provides the following:
+	// 1) Multiple plz instances can run concurrently.
+	// 2) If another process tries to obtain an exclusive repo lock, it will have to wait until any existing repo locks are released in other processes.
+	// This is useful for things like when plz tries to download and update itself.
+	// 3) A new plz process will have to wait to acquire its shared repo lock, if there's already an existing process with an exclusive repo lock.
+	core.AcquireSharedRepoLock()
```

Comment:
I'm not sure i fully understand why we need a shared lock. If I understand how this is implemented, a `plz build` would acquire the shared lock, and then a `plz update` would have to wait for the build to finish before downloading the new Please version. Is downloading Please while building problematic?

Reply:
As discussed IRL, I'm not really sure whether it would problematic. The current logic does rely on it so I kept it. Although only `plz update` uses an exclusive lock for now, there might be other code paths in the future that require a exclusive lock, so this logic is already accounting for that.